coqlspclient/coq_file.py

```diff
@@ -115,6 +115,17 @@ def __validate(self):
                 return
         self.is_valid = True
 
+    @staticmethod
+    def get_id(id: List) -> str:
+        name = ""
+        if id[0] == "Ser_Qualid" and id[1][0] == "DirPath" and id[2][0] == "Id":
+            for dir_el in id[1][1]:
+                name += dir_el[1] + "."
+            name += id[2][1]
+        elif id[0] == "Id":
+            name = id[1]
+        return name
```

Comment:
I think you can assume that a "Ser_Qualid" will always have a "DirPath" and "Id". The one-liner [here](https://github.com/sr-lab/coq-lsp-pyclient/blob/type_class/coqlspclient/proof_state.py#L239) never crashed and does the same. Maybe replace the code here and call `get_id` in the linked file. Personally, I like the one-liner, but it's fine if you leave it as is :slightly_smiling_face: 
```suggestion
    def get_id(id: List) -> str:
        if id[0] == "Ser_Qualid":
            return ".".join([l[1] for l in id[1][1][::-1]] + [id[2][1]])
        elif id[0] == "Id":
            return id[1]
        return ""
```